Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp

```diff
@@ -43,11 +43,38 @@ namespace Platform
         }
         return physicalDeviceList;
     }
+    
+    float GetRefreshRate()
+    {
+        CGDirectDisplayID display = CGMainDisplayID();
+        CGDisplayModeRef currentMode = CGDisplayCopyDisplayMode(display);
+        return CGDisplayModeGetRefreshRate(currentMode);
+    }
 
-    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval)
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
     {
-        AZ_UNUSED(syncInterval);
-        [mtlCommandBuffer presentDrawable:drawable];
+        bool hasPresentAfterMinimumDurationAPI = false;
+        bool framePresented = false;
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
+        
+#if defined(__MAC_11_0)
+        if(@available(iOS 14.0, macOS 11.0, *))
```

Comment:
Why is this extra branch needed given the header guard above (and why does it check ios availability)?

*NOTE: I don't know much objective-C so I may just be reading this incorrectly*

Reply:
Branch is needed because we support running on iOS13.

Reply:
The first branch is protect against compilation and the second is to check if at runtime you are targeting the correct OS. Actually I will remove the ios check. I forgot to take that out.